drivers/soundwire/intel.c

```diff
@@ -384,13 +392,15 @@ static void intel_shim_wake(struct sdw_intel *sdw, bool wake_enable)
 	unsigned int link_id = sdw->instance;
 	u16 wake_en, wake_sts;
 
+	mutex_lock(sdw->link_res->shim_lock);
+	wake_en = intel_readw(shim, SDW_SHIM_WAKEEN);
+
 	if (wake_enable) {
 		/* Enable the wakeup */
-		intel_writew(shim, SDW_SHIM_WAKEEN,
-			     (SDW_SHIM_WAKEEN_ENABLE << link_id));
+		wake_en &= (SDW_SHIM_WAKEEN_ENABLE << link_id);
```

Comment:
here, should be |=,   I also have a PR to fix this https://github.com/thesofproject/linux/pull/1579

Reply:
ah yes, this is stupid, thanks for noticing this.